src/main/kotlin/app/revanced/patches/youtube/layout/hide/player/overlay/patch/HidePlayerOverlayPatch.kt

```diff
@@ -1,46 +1,77 @@
+            it.name == "scrim_overlay"
+        }.id
+        const val INTEGRATIONS_CLASS_DESCRIPTOR = "Lapp/revanced/integrations/patches/HidePlayerOverlayPatch;"
+    }
+
+    override fun execute(context: BytecodeContext): PatchResult {
+        SettingsPatch.PreferenceScreen.LAYOUT.addPreferences(
+            SwitchPreference(
+                "revanced_hide_player_overlay",
+                StringResource("revanced_hide_player_overlay_title", "Hide player overlay"),
+                false,
+                StringResource("revanced_hide_player_overlay_summary_on", "Player overlay is hidden"),
+                StringResource("revanced_hide_player_overlay_summary_off", "Player overlay is shown")
+            )
+        )
+
+        context.classes.forEach { classDef ->
+            classDef.methods.forEach { method ->
+                with(method.implementation) {
+                    this?.instructions?.forEachIndexed { index, instruction ->
+                        when (instruction.opcode) {
+                            Opcode.CONST -> {
+                                when ((instruction as Instruction31i).wideLiteral) {
+                                    scrimOverlayId -> { // player overlay filter
+                                        val insertIndex = index + 3
+                                        val invokeInstruction = instructions.elementAt(insertIndex)
+                                        if (invokeInstruction.opcode != Opcode.CHECK_CAST) return@forEachIndexed
 
-        context.xmlEditor[RESOURCE_FILE_PATH].use { editor ->
-            editor.file.getElementsByTagName("FrameLayout").item(0).childNodes.apply {
-                for (i in 1 until length) {
-                    val view = item(i)
-                    if (
-                        view.attributes.getNamedItem("android:id")
-                            ?.nodeValue
-                            ?.endsWith("scrim_overlay") == true
-                    ) {
-                        attributes.forEach {
-                            view.attributes.getNamedItem("android:layout_$it").nodeValue = "0.0dip"
+                                        val mutableMethod = context.proxy(classDef).mutableClass.findMutableMethodOf(method)
+                                        val viewRegister = (invokeInstruction as Instruction21c).registerA
+
+                                        mutableMethod.addInstruction(
+                                            insertIndex + 1,
+                                            "invoke-static {v$viewRegister}, $INTEGRATIONS_CLASS_DESCRIPTOR->hidePlayerOverlay(Landroid/widget/ImageView;)V"
+                                        )
+                                    }
+                                }
+                            }
+                            else -> return@forEachIndexed
                         }
-                        break
                     }
                 }
             }
         }
```

Comment:
This should be refactored to a `MethodFingerprint`, instead of manually looping through all classes. Several patches use a resource id to resolve, which can be taken as a reference.